project/src/components/app/app.tsx

```diff
@@ -1,11 +1,35 @@
+import { BrowserRouter, Routes, Route } from 'react-router-dom';
+
 import MainPage from '../../pages/main-page/main-page';
+import Favorites from '../../pages/favorites/favorites';
+import Propepty from '../../pages/property/property';
+import Login from '../../pages/login/login';
+import NotFound from '../../pages/not-found/not-found';
+
+import PrivateOutlet from '../private-route/private-route';
+import { AppRoute, AuthorizationStatus } from '../../const';
 
 interface AppProps {
   numberOfPlaces: number;
 }
 
 function App({ numberOfPlaces }: AppProps): JSX.Element {
-  return <MainPage numberOfPlaces={numberOfPlaces} isEmpty={false} />;
+  return (
+    <BrowserRouter>
+      <Routes>
+        <Route index element={<MainPage numberOfPlaces={numberOfPlaces} isEmpty={false} />} />
+        <Route path={AppRoute.FAVORITES} element={<PrivateOutlet authorizationStatus={AuthorizationStatus.NO_AUTH} />}>
+          <Route path="" element={<Favorites isEmpty={false} />} />
```

Comment:
path надо перенести ниже. Преимущество решения с Outlet то что ты можешь описать несколько маршрутов закрытых в одном роуте. Например если нам нужно закрыть еще и страницу Login (поправь только редирект, если будешь тестить)
```suggestion
        <Route  element={<PrivateOutlet authorizationStatus={AuthorizationStatus.NO_AUTH} />}>
          <Route path={AppRoute.FAVORITES} element={<Favorites isEmpty={false} />} />
          <Route path={AppRoute.LOGIN} element={<Login />} />
```